service/awsconfig/v10/resource/s3bucket/error.go

```diff
@@ -26,8 +28,12 @@ func IsBucketNotFound(err error) bool {
 	if !ok {
 		return false
 	}
-	// TODO Find constant in the Go SDK for the error code.
-	if aerr.Code() == "NotFound" {
+	if aerr.Code() == s3.ErrCodeNoSuchBucket {
```

Comment:
sorry I dont get you. What do you mean?

Reply:
`aerr` is the result of a type assertion above. Base for the assertion is `err`. Base should be `microerror.Cause(err)` because only the cause is the actual error we want to check against. This here can result to nasty side effects.

Reply:
hmm but then it should be like
```
var notSuchBucket = microerror.New(s3.ErrCodeNoSuchBucket)

// IsBucketNotFound asserts bucket not found error from upstream's API code.
func IsBucketNotFound(err error) bool {
	return microerror.Cause(err) == notSuchBucket
}
```

Reply:
IMO this will not work because the S3 error you want to check against is also wrapped. It is ok to have our native error and other stuff being checked in the matcher functions.